### DISCUSSION TITLE
Webpack 5.3.2 fails to properly compile changes when using "webpack --mode development --watch"

### DISCUSSION THREAD

**Discussion by william-herring** [webpack/webpack#15000] [Q&A] [answered]
# Bug report

**What is the current behavior?**
When running "webpack --mode development --watch" to watch for changes in my code, webpack claims to compile successfully despite it being visible on a page display that no changes were applied. I have experienced this multiple times on different projects, and it only starts to occur after around 200 compilations (estimate). No errors occur, terminal output is logging changes as compiled. I suspect this may have something to do with the cache but am not totally sure.

**If the current behavior is a bug, please provide the steps to reproduce.**
<a href="https://github.com/william-herring/swapit/tree/main/client">Repo</a>

Server-side rendered React app with following config files:

package.json
```json
{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "webpack --mode development --watch",
    "build": "webpack --mode production"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@babel/preset-env": "^7.12.1",
    "@babel/preset-react": "^7.12.1",
    "autoprefixer": "^10.4.0",
    "babel-loader": "^8.1.0",
    "css-loader": "^6.5.1",
    "html-webpack-plugin": "^5.5.0",
    "node-sass": "^7.0.0",
    "postcss": "^8.4.5",
    "postcss-cli": "^9.1.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "sass-loader": "^12.4.0",
    "style-loader": "^3.3.1",
    "tailwindcss": "^3.0.2",
    "terser-webpack-plugin": "^5.0.3",
    "webpack": "^5.3.2",
    "webpack-cli": "^4.1.0"
  },
  "dependencies": {
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "postcss-loader": "^6.2.1",
    "postcss-preset-env": "^7.0.1",
    "react-router-dom": "^5.2.0"
  }
}
```

webpack.config.js
```js
const path = require("path");
const webpack = require("webpack");

module.exports = {
    entry: "./src/index.js",
    output: {
        path: path.resolve(__dirname, "./static/"),
        filename: "[name].js",
    },
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /node_modules/,
                use: {
                    loader: "babel-loader",
                },
            },
            {
                test: /\.css$/,
                use: [
                    'style-loader',
                    'css-loader',
                    'postcss-loader',
                ],
            }
        ],
    },
    optimization: {
        minimize: true,
    },
    plugins: [
        new webpack.DefinePlugin({
            'process.env.NODE_ENV' : JSON.stringify('development')
        })
    ],
};
```

Please see main.js and other code in repository linked at top.

**What is the expected behavior?**
Code should re-compile and apply new changes.

**Other relevant information:**
webpack version:  5.3.2
Node.js version: 16.13.0
Operating System: MacOS Monterey (12.0.1)
Additional tools: React 17.0.1

**Comment by alexander-akait** [comment, marked as answer]
Please update webpack version

**Comment by alexander-akait** [comment]
Also

> When running "webpack --mode development --watch" to watch for changes in my code, webpack claims to compile successfully despite it being visible on a page display that no changes were applied.

please clarify what do you mean. Does it happen for JS/CSS/HTML? Why did you decide that they did not appear and is it not browser cache or network latency? What files will it go to, where will they be located? I tried to reproduce your repo and after 1000 compilation all works fine? How long does it take for this to happen?

**Comment by william-herring** [comment]
Update: really silly mistake, honestly. Just needed to update webpack version. Sorry about that. @alexander-akait, In case the details are still needed, this issue occurred with *any* change to the codebase.

**Reply by alexander-akait** [reply]
After update no problems?

**Reply by william-herring** [reply]
Yep. You can close this now if you wish.
